starknet_py/utils/data_transformer/data_transformer.py

```diff
@@ -361,5 +371,26 @@ def _remove_array_lengths(type_by_name: dict) -> dict:
         return {
             k: v
             for k, v in type_by_name.items()
-            if not DataTransformer._is_array_len(k, v, type_by_name=type_by_name)
+            if not CairoSerializer._is_array_len(k, v, type_by_name=type_by_name)
         }
+
+
+class FunctionCallSerializer:
+    """
+    Transforms function call data from python to Cairo format and back.
+    """
+
+    def __init__(self, abi: ABIFunctionEntry, identifier_manager: IdentifierManager):
+        self.structure_transformer = CairoSerializer(identifier_manager)
+        self.abi = abi
+
+    def from_python(self, *args, **kwargs) -> (List[int], Dict[str, List[int]]):
+        return self.structure_transformer.from_python(
+            self.abi["inputs"], *args, **kwargs
+        )
+
+    def to_python(self, values: CairoData) -> NamedTuple:
+        return self.structure_transformer.to_python(self.abi["outputs"], values)
+
+
+DataTransformer = FunctionCallSerializer
```

Comment:
I don't like it, I think we should just rename every usage of DataTransformer.

Or we could do some kind of deprecation [like this ](https://stackoverflow.com/questions/9008444/how-to-warn-about-class-name-deprecation), and remove it in the future.

Reply:
@Solpatium what do you think?

Reply:
Yes, please add a deprecation warning.